src/main/java/io/citrine/jcc/search/core/result/BaseMultiSearchResult.java

```diff
@@ -14,10 +14,11 @@
  * Base class for all multi-search results.
  *
  * @param <T> Type of the results.
+ * @param <U> Type of the atomic search result.
  * @author Kyle Michel
  */
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public abstract class BaseMultiSearchResult<T> implements Iterable<T> {
+public abstract class BaseMultiSearchResult<T, U extends BaseMultiSearchResultElement<T>> implements Iterable<U> {
```

Comment:
Why aren't we setting the type parameters in the definition of `setTook`?  (it won't let me comment there because you didn't edit it).